lib/rubocop/cop/variable_force/branch.rb

```diff
@@ -127,7 +127,7 @@ def ==(other)
           alias_method :eql?, :==
 
           def hash
-            [control_node.object_id, control_node.object_id].hash
+            [control_node.object_id, control_node.object_id].hash # rubocop:disable Lint/HashMissingClass
```

Comment:
There's a lot to discuss/investigate here

- We introduced a bug in #10441 and replaced `child_node` with a second `control_node`, which needs fixing
- We _might not_ need to call `.object_id` on the nodes, but maybe there's a reason not captured in tests
- We _probably_ should include `self.class`, but again maybe there's a reason not captured in tests

I decided I'll open a separate PR for discussing/addressing these items, to keep this PR focussed.